devdoc/CodeReviews.md

```diff
@@ -0,0 +1,55 @@
+Some general guidelines for code reviews.
+
+Code reviews have a few different goals:
+* Catch architectural or design problems
+* Catch bugs early (incorrect behaviour, inefficiencies, security issues)
+* Ensure the code is readable and maintainable.
+* A opportunity for training/passing on information.  
+  For example providing information about how the current system works, functionality that is already available or suggestions of other approaches the developer may not have thought of.
+
+Some general comments:
+- Code reviewers should be explicit and clearly describe the problem, including what change is expected if not obvious. Don’t assume the contributor has same understanding/view as reviewer.
+- If a comment is not clear the contributor should ask for clarification rather than wasting time trying to second-guess the reviewer.
+- Contributors should feel free to push back if they consider comments are too picky.  The reviewer can either agree, or provide reasons why they consider it to be an issue.
+- Sometimes a significant design problem means the rest of the code has not been reviewed in detail.  If this is the case it should be explicitly noted.
+- Repeated issue.  The reviewer is free to comment on every instance of a repeated issue, but a simple annotation should alert contributor to address appropriately eg: [Please address all instances of this issue]
+- The contributor should respond to comment if it isn't obvious where/how they have been addressed (but no need to acknowledge typo/indentation/etc)
+- Only the reviewer should mark issues as resolved using the Github resolve conversation button.
+- Both reviewers and contributors should respond in a timely manner - don't leave it for days.  It destroys the flow of thought and conversation.
+- Watch out for collapsed conversations.  If there are large numbers of comments GitHib will collapse them which can make them easy to miss.
+- If there are large number of comments > 100, it may be better to close the PR and open a new one.
+- Submit any changes as extra commits so that it is clear to the reviewer what has changed, and avoids them having to re-review everything.  Please do not squash them until the reviewers approve the PR.  A few exceptions to this are if the only comments that the reviewer makes are indentation/typos, if the PR is only a couple of lines, or the PR is completely rewritten in response to the review.
```

Comment:
From a recent experience with one of my own prs, I think the first exception is wrong - best not to squash even if the changes are trivial because it still makes them harder to spot.